engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractor.scala

```diff
@@ -85,15 +86,35 @@ object ProcessDefinitionExtractor {
 
   private def extractCustomTransformerData(objectWithMethodDef: ObjectWithMethodDef) = {
     val transformer = objectWithMethodDef.obj.asInstanceOf[CustomStreamTransformer]
+    val contextTransformationOpt = extractContextTransformation(objectWithMethodDef)
     val queryNamesAnnotation = objectWithMethodDef.methodDef.annotations.flatMap(_.cast[QueryableStateNames])
     val queryNames = queryNamesAnnotation.flatMap(_.values().toList).toSet
-    CustomTransformerAdditionalData(queryNames, transformer.clearsContext, transformer.canHaveManyInputs)
+    CustomTransformerAdditionalData(queryNames, contextTransformationOpt, transformer.clearsContext, transformer.canHaveManyInputs)
+  }
+
+  private def extractContextTransformation(objectWithMethodDef: ObjectWithMethodDef): Option[ContextTransformation] = {
```

Comment:
why AddsNewVariable and ProcessContext not special case of TransformContext? Let's try to get rid of special cases?